--- kernel/yosys.cc
+++ kernel/yosys.cc
@@ -930,13 +930,12 @@ void init_share_dirname()
 		yosys_share_dirname = proc_share_path;
 		return;
 	}
-#    ifdef YOSYS_DATDIR
-	proc_share_path = YOSYS_DATDIR "/";
+	const char* env_datdir = std::getenv("YOSYS_DATDIR");
+	proc_share_path = env_datdir != nullptr ? env_datdir : "./";
 	if (check_file_exists(proc_share_path, true)) {
 		yosys_share_dirname = proc_share_path;
 		return;
 	}
-#    endif
 #  endif
 }
 #endif
