--- boost.BUILD
+++ boost.BUILD
@@ -1927,13 +1927,13 @@
     exclude_src = ["libs/stacktrace/src/*.cpp"],
     linkopts = select({
         ":linux_ppc": [
-            "-lbacktrace -ldl",
+            "-ldl",
         ],
         ":linux_x86_64": [
-            "-lbacktrace -ldl",
+            "-ldl",
         ],
         ":linux_aarch64": [
-            "-lbacktrace -ldl",
+            "-ldl",
         ],
         "//conditions:default": [],
     }),
@@ -1946,6 +1946,7 @@
         ":predef",
         ":static_assert",
         ":type_traits",
+        "@com_github_libbacktrace//:libbacktrace",
     ],
 )
 
