diff --git a/cells/lpflow_lsbuf_lh_hl_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_1.lef b/cells/lpflow_lsbuf_lh_hl_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_1.lef
index 6049eb25..ec57a8ff 100644
--- a/cells/lpflow_lsbuf_lh_hl_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_1.lef
+++ b/cells/lpflow_lsbuf_lh_hl_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_1.lef
@@ -14,8 +14,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0

-VERSION 5.5 ;
-NAMESCASESENSITIVE ON ;
+VERSION 5.6 ;
 BUSBITCHARS "[]" ;
 DIVIDERCHAR "/" ;
 MACRO sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_1
diff --git a/cells/lpflow_lsbuf_lh_hl_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_2.lef b/cells/lpflow_lsbuf_lh_hl_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_2.lef
index 57b82226..cc729edc 100644
--- a/cells/lpflow_lsbuf_lh_hl_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_2.lef
+++ b/cells/lpflow_lsbuf_lh_hl_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_2.lef
@@ -14,8 +14,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0

-VERSION 5.5 ;
-NAMESCASESENSITIVE ON ;
+VERSION 5.6 ;
 BUSBITCHARS "[]" ;
 DIVIDERCHAR "/" ;
 MACRO sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_2
diff --git a/cells/lpflow_lsbuf_lh_hl_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_4.lef b/cells/lpflow_lsbuf_lh_hl_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_4.lef
index 0325315e..f4ad4816 100644
--- a/cells/lpflow_lsbuf_lh_hl_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_4.lef
+++ b/cells/lpflow_lsbuf_lh_hl_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_4.lef
@@ -14,8 +14,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0

-VERSION 5.5 ;
-NAMESCASESENSITIVE ON ;
+VERSION 5.6 ;
 BUSBITCHARS "[]" ;
 DIVIDERCHAR "/" ;
 MACRO sky130_fd_sc_hd__lpflow_lsbuf_lh_hl_isowell_tap_4
diff --git a/cells/lpflow_lsbuf_lh_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_1.lef b/cells/lpflow_lsbuf_lh_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_1.lef
index c87e7b12..bf0145f4 100644
--- a/cells/lpflow_lsbuf_lh_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_1.lef
+++ b/cells/lpflow_lsbuf_lh_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_1.lef
@@ -14,8 +14,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0

-VERSION 5.5 ;
-NAMESCASESENSITIVE ON ;
+VERSION 5.6 ;
 BUSBITCHARS "[]" ;
 DIVIDERCHAR "/" ;
 MACRO sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_1
diff --git a/cells/lpflow_lsbuf_lh_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_2.lef b/cells/lpflow_lsbuf_lh_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_2.lef
index add5ba3e..d521bd82 100644
--- a/cells/lpflow_lsbuf_lh_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_2.lef
+++ b/cells/lpflow_lsbuf_lh_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_2.lef
@@ -14,8 +14,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0

-VERSION 5.5 ;
-NAMESCASESENSITIVE ON ;
+VERSION 5.6 ;
 BUSBITCHARS "[]" ;
 DIVIDERCHAR "/" ;
 MACRO sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_2
diff --git a/cells/lpflow_lsbuf_lh_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_4.lef b/cells/lpflow_lsbuf_lh_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_4.lef
index 14aa71a5..b623e271 100644
--- a/cells/lpflow_lsbuf_lh_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_4.lef
+++ b/cells/lpflow_lsbuf_lh_isowell_tap/sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_4.lef
@@ -14,8 +14,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0

-VERSION 5.5 ;
-NAMESCASESENSITIVE ON ;
+VERSION 5.6 ;
 BUSBITCHARS "[]" ;
 DIVIDERCHAR "/" ;
 MACRO sky130_fd_sc_hd__lpflow_lsbuf_lh_isowell_tap_4
diff --git a/cells/tap/sky130_fd_sc_hd__tap_1.lef b/cells/tap/sky130_fd_sc_hd__tap_1.lef
index 5b2e0545..e6fcbb20 100644
--- a/cells/tap/sky130_fd_sc_hd__tap_1.lef
+++ b/cells/tap/sky130_fd_sc_hd__tap_1.lef
@@ -14,8 +14,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0

-VERSION 5.5 ;
-NAMESCASESENSITIVE ON ;
+VERSION 5.6 ;
 BUSBITCHARS "[]" ;
 DIVIDERCHAR "/" ;
 MACRO sky130_fd_sc_hd__tap_1
diff --git a/cells/tap/sky130_fd_sc_hd__tap_2.lef b/cells/tap/sky130_fd_sc_hd__tap_2.lef
index 7ee145fa..d51cbb8a 100644
--- a/cells/tap/sky130_fd_sc_hd__tap_2.lef
+++ b/cells/tap/sky130_fd_sc_hd__tap_2.lef
@@ -14,8 +14,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0

-VERSION 5.5 ;
-NAMESCASESENSITIVE ON ;
+VERSION 5.6 ;
 BUSBITCHARS "[]" ;
 DIVIDERCHAR "/" ;
 MACRO sky130_fd_sc_hd__tap_2
diff --git a/cells/tapvgnd/sky130_fd_sc_hd__tapvgnd_1.lef b/cells/tapvgnd/sky130_fd_sc_hd__tapvgnd_1.lef
index c4ad8366..3e8176e3 100644
--- a/cells/tapvgnd/sky130_fd_sc_hd__tapvgnd_1.lef
+++ b/cells/tapvgnd/sky130_fd_sc_hd__tapvgnd_1.lef
@@ -14,8 +14,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0

-VERSION 5.5 ;
-NAMESCASESENSITIVE ON ;
+VERSION 5.6 ;
 BUSBITCHARS "[]" ;
 DIVIDERCHAR "/" ;
 MACRO sky130_fd_sc_hd__tapvgnd_1
diff --git a/cells/tapvgnd2/sky130_fd_sc_hd__tapvgnd2_1.lef b/cells/tapvgnd2/sky130_fd_sc_hd__tapvgnd2_1.lef
index 7e6d1238..c6ad97d6 100644
--- a/cells/tapvgnd2/sky130_fd_sc_hd__tapvgnd2_1.lef
+++ b/cells/tapvgnd2/sky130_fd_sc_hd__tapvgnd2_1.lef
@@ -14,8 +14,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0

-VERSION 5.5 ;
-NAMESCASESENSITIVE ON ;
+VERSION 5.6 ;
 BUSBITCHARS "[]" ;
 DIVIDERCHAR "/" ;
 MACRO sky130_fd_sc_hd__tapvgnd2_1
diff --git a/cells/tapvpwrvgnd/sky130_fd_sc_hd__tapvpwrvgnd_1.lef b/cells/tapvpwrvgnd/sky130_fd_sc_hd__tapvpwrvgnd_1.lef
index 02d7c833..a8b17faa 100644
--- a/cells/tapvpwrvgnd/sky130_fd_sc_hd__tapvpwrvgnd_1.lef
+++ b/cells/tapvpwrvgnd/sky130_fd_sc_hd__tapvpwrvgnd_1.lef
@@ -14,8 +14,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0

-VERSION 5.5 ;
-NAMESCASESENSITIVE ON ;
+VERSION 5.6 ;
 BUSBITCHARS "[]" ;
 DIVIDERCHAR "/" ;
 MACRO sky130_fd_sc_hd__tapvpwrvgnd_1
