diff --git a/unix/tclUnixInit.c b/unix/tclUnixInit.c
index aac8a8d..42eb45d 100644
--- a/unix/tclUnixInit.c
+++ b/unix/tclUnixInit.c
@@ -11,6 +11,7 @@
 #include "tclInt.h"
 #include <stddef.h>
 #include <locale.h>
+#include <unistd.h>
 #ifdef HAVE_LANGINFO
 #   include <langinfo.h>
 #   ifdef __APPLE__
@@ -547,6 +548,15 @@ TclpInitLibraryPath(
 	    Tcl_ListObjAppendElement(NULL, pathPtr, objPtr);
 	}
     }
+
+	const size_t path_length = 5;
+	char binary_location[path_length] = {'\0'};
+	readlink("/proc/self/exe", binary_location, sizeof(binary_location)-1);
+	if(binary_location[path_length-2] != '\0') {
+		printf("[fatal] TCL_LIBRARY path is longer than %zu characters!\n", path_length);
+		exit(-1);
+	}
+	binary_location[path_length-2] = '\0';
     Tcl_DStringFree(&buffer);

     *encodingPtr = Tcl_GetEncoding(NULL, NULL);
